src/dialects/mssql/query-generator.js

```diff
@@ -342,20 +340,27 @@ export class MsSqlQueryGenerator extends AbstractQueryGenerator {
   }
 
   removeColumnQuery(tableName, attributeName, options) {
-    if (options) {
+    options = options || {};
+
+    const dbVersion = this.sequelize.options.databaseVersion;
+    const isGreaterThanSQLServer2016 = semver.valid(dbVersion) && semver.gte(dbVersion, '13.0.0');
```

Comment:
We only support SQL Server 2017+ in v7, so you can drop this check

Actually looks like it could become SQL Server 2019+ since support for 2017 ended 4 weeks ago